src/stamp.cr

```diff
@@ -0,0 +1,81 @@
+class Hashcash::Stamp
+  STAMP_VERSION = 1
+  getter version, bits, date, resource, ext, rand
+  property counter
```

Comment:
i'd make this a getter, as you probably don't want to allow external mutation, and you're using `@counter` in `update_counter`